docs/guide/troubleshooting.md

```diff
@@ -121,6 +121,16 @@ You will need to access the file with `http` protocol. The easiest way to achiev
 
 ## Others
 
+### Module externalized for browser compatibility
+
+When you use a Node.js module, Vite will output the following warning.
+
+> Module "fs" has been externalized for browser compatibility. Cannot access "fs.readFile" in client code.
+
+This is because Vite does not automatically polyfill Node.js modules.
+
+We recommend avoiding Node.js modules for browser code to reduce the bundle size. Although you can add polyfills manually.
```

Comment:
```suggestion
We recommend avoiding Node.js modules for browser code to reduce the bundle size, although you can add polyfills manually. If the module is imported from a third-party library (that's meant to be used in the browser), it's advised to report the issue to the respective library.
```

I think we could add some notes if it happens in a dep

Reply:
Looks nice. Thanks!